lib/Fmt_ast.ml

```diff
@@ -332,6 +332,17 @@ let fmt_constant c ~loc ?epi const =
           fmt_words ~epi mode curr $ opt next fmt_next
         in
         let lines = String.split ~on:'\n' s in
+        let lines =
+          if break_on_newlines then lines
+          else
+            (* linebreaks are merged with the preceding line when possible
+               instead of having a blank line in the list *)
+            List.foldi lines ~init:[] ~f:(fun i acc -> function
+              | "" when i < List.length lines - 1 -> (
```

Comment:
Probably best to precompute `List.length lines` to avoid complexity potentially quadratic in number of lines.

Reply:
Oh right, I will fix this